#
# Submitted-By: Robert Schwebel, 2007-01-03
# Committed-By: Robert Schwebel
#
# Error:
#
# freetype-2.2.1/objs/apinames: cannot execute binary file
#
# Description:
#
# The make system does not differentiate between tools for the "host"
# and "build" system, so we introduce CC_FOR_BUILD here.
#
# State:
#
# not reported yet
#

Index: builds/exports.mk
===================================================================
--- a/builds/exports.mk.orig
+++ b/builds/exports.mk
@@ -51,7 +51,7 @@ ifneq ($(EXPORTS_LIST),)
   APINAMES_EXE := $(OBJ_DIR)/apinames$E
 
   $(APINAMES_EXE): $(APINAMES_SRC)
-	  $(CCexe) $(TE)$@ $<
+	  $(CC_FOR_BUILDexe) $(TE)$@ $<
 
   .PHONY: symbols_list
 
Index: builds/unix/configure.ac
===================================================================
--- a/builds/unix/configure.ac.orig
+++ b/builds/unix/configure.ac
@@ -33,6 +33,14 @@ AC_CANONICAL_TARGET
 AC_PROG_CC
 AC_PROG_CPP
 
+if test "x$cross_compiling" = "xno"; then
+  CC_FOR_BUILD=${CC_FOR_BUILD-${CC-gcc}}
+else
+  CC_FOR_BUILD=${CC_FOR_BUILD-gcc}
+fi
+
+AC_SUBST(CC_FOR_BUILD)
+
 
 # get compiler flags right
 
Index: builds/unix/unix-cc.in
===================================================================
--- a/builds/unix/unix-cc.in.orig
+++ b/builds/unix/unix-cc.in
@@ -13,6 +13,7 @@
 
 
 CC           := @CC@
+CC_FOR_BUILD := @CC_FOR_BUILD@
 COMPILER_SEP := $(SEP)
 
 LIBTOOL ?= $(BUILD_DIR)/libtool
@@ -80,8 +81,10 @@ ANSIFLAGS := @XX_ANSIFLAGS@
 # C compiler to use -- we use libtool!
 #
 #
-CCraw := $(CC)
-CC    := $(LIBTOOL) --mode=compile $(CCraw)
+CCraw           := $(CC)
+CC              := $(LIBTOOL) --mode=compile $(CCraw)
+CC_FOR_BUILDraw := $(CC_FOR_BUILD)
+CC_FOR_BUILD    := $(LIBTOOL) --mode=compile $(CC_FOR_BUILDraw)
 
 # Linker flags.
 #
@@ -90,8 +93,9 @@ LDFLAGS := @LDFLAGS@
 
 # export symbols (XXX: HOW TO DEAL WITH CROSS COMPILATION ?)
 #
-EXPORTS_LIST := $(OBJ_DIR)/ftexport.sym
-CCexe        := $(CCraw)   # used to compile "apinames" only
+EXPORTS_LIST    := $(OBJ_DIR)/ftexport.sym
+CCexe           := $(CCraw)   # used to compile "apinames" only
+CC_FOR_BUILDexe := $(CC_FOR_BUILDraw)
 
 
 # Library linking
